tests/ga/test_extension_telemetry.py

```diff
@@ -0,0 +1,533 @@
+import contextlib
+import glob
+import json
+import os
+import random
+import re
+import shutil
+import string
+import uuid
+from collections import defaultdict
+
+from mock import MagicMock, Mock, patch
+
+from azurelinuxagent.common import logger, conf
+from azurelinuxagent.common.event import EVENTS_DIRECTORY
+from azurelinuxagent.common.exception import InvalidExtensionEventError
+from azurelinuxagent.common.logger import Logger
+from azurelinuxagent.common.protocol.util import ProtocolUtil
+from azurelinuxagent.common.protocol.wire import event_param_to_v1
+from azurelinuxagent.common.telemetryevent import TelemetryEventParam, GuestAgentGenericLogsSchema, \
+    CommonTelemetryEventSchema
+from azurelinuxagent.common.utils import fileutil, textutil
+from azurelinuxagent.ga.extension_telemetry import get_extension_telemetry_handler, ExtensionEventSchema
+from tests.protocol.mocks import mock_wire_protocol, HttpRequestPredicates, MockHttpResponse
+from tests.protocol.mockwiredata import DATA_FILE
+from tests.tools import AgentTestCase, clear_singleton_instances, data_dir, skip_if_predicate_true
+
+
+class TestExtensionTelemetryHandler(AgentTestCase, HttpRequestPredicates):
+
+    _TEST_DATA_DIR = os.path.join(data_dir, "events", "extension_events")
+    _WELL_FORMED_FILES = os.path.join(_TEST_DATA_DIR, "well_formed_files")
+    _MALFORMED_FILES = os.path.join(_TEST_DATA_DIR, "malformed_files")
+    _MIX_FILES = os.path.join(_TEST_DATA_DIR, "mix_files")
+
+    # To make tests more versatile, include this key in a test event to mark that event as a bad event.
+    # This event will then be skipped and will not be counted as a good event. This is purely for testing purposes,
+    # we use the good_event_count to validate the no of events the agent actually sends to Wireserver.
+    # Eg: {
+    #         "EventLevel": "INFO",
+    #         "Message": "Starting IaaS ScriptHandler Extension v1",
+    #         "Version": "1.2.3",
+    #         "TaskName": "Extension Info",
+    #         "EventPid": "5676",
+    #         "EventTid": "1",
+    #         "OperationId": "e1065def-7571-42c2-88a2-4f8b4c8f226d",
+    #         "TimeStamp": "2019-12-12T01:11:38.2298194Z",
+    #         "BadEvent": true
+    #     }
+    BAD_EVENT_KEY = 'BadEvent'
+
+    def setUp(self):
+        AgentTestCase.setUp(self)
+        prefix = "ExtensionTelemetryUnitTest"
+        logger.DEFAULT_LOGGER = Logger(prefix=prefix)
```

Comment:
this should be reverted to the original value on cleanup

Reply:
Ohh didnt know that, done!